Omnikit status command errors

Lytrix opened the pull request:
Started working on updating the getStatusCommand
These are the possible status request values, which will also be used in the status error. (I've started on that as well)
And a small fix to have the bolus tests back.

Let me know if you want to merge these small steps. I can also continue with adding more stuff in for the status error responses as well and then do the PR later.

Lytrix committed:
'Fixed missing Foundation and availability of Bolus tests in Xcode'

Lytrix committed:
Updated statusTypes

Lytrix committed:
added status test and fixed typo

ps2 commented on OmniKit/MessageTransport/MessageBlocks/GetStatusCommand.swift:
```diff
@@ -8,44 +8,49 @@
-}
-
-struct GetStatusCommand : MessageBlock {
+    public enum GetStatusCommandError: Error {
```
This can be removed if you're using MessageError.unknownValue instead.

Lytrix committed:
removed obsolete unknown requesttype function

Lytrix committed:
added more statusFaultError tests and values

Lytrix committed:
Added more basic test values to StatusError Type2

Lytrix committed:
Added to statusError:
- insulinRemaining
- timeActive
- secondaryLoggedFaultEvent
- logEventError
- New test for testing one Fault during deactivation

Lytrix committed:
Added remaining values:
- insulinStateCorruptionDuringErrorLogging
- progressAtFirstLoggedFaultEvent
- recieverLowGain
- radioRSSI
- progressAtFirstLoggedFaultEventCheck
- Fixed wrong low bit selection  on radio
Removed StatusConfigAlarms test for now, must use separate route to evaluate.

ps2 commented on OmniKit/MessageTransport/MessageBlocks/StatusError.swift:
```diff
@@ -9,16 +9,137 @@
+        case tankPowerActivated = 1
+        case tankFillCompleted = 2
+        case pairingSuccess = 3
+        case Purging = 4
```
enums should begin lowercase

ps2 commented on OmniKit/MessageTransport/MessageBlocks/StatusError.swift:
```diff
@@ -9,16 +9,137 @@
+    public let insulinNotDelivered: Double
+    public let podMessageCounter: UInt8
+    public let unknownPageCode: Double
+    public let origionalLoggedFaultEvent: UInt8
```
spelling: origional -> original

ps2 commented on OmniKit/MessageTransport/MessageBlocks/StatusError.swift:
```diff
@@ -9,16 +9,137 @@
+        case primingCannula = 7
+        case runningNormal = 8
+        case runningLessThen50ULeftInReservoir = 9
+        case OnenotUsedButin33 = 10
```
.oneNotUsedButIn33

ps2 commented on OmniKit/MessageTransport/MessageBlocks/StatusError.swift:
```diff
@@ -9,16 +9,137 @@
+        case podInactive = 15  // ($1C Deactivate Pod or packet header mismatch)
+    }
+
+    public enum DeliveryInProgressType: UInt8 {
```
Looking on the wiki, it looks like this is a bitfield.  If that is the case, this should be an OptionSet

ps2 commented on OmniKit/MessageTransport/MessageBlocks/StatusError.swift:
```diff
@@ -9,16 +9,137 @@
+        
+        self.recieverLowGain = encodedData[20] >> 4
+        
+        self.radioRSSI =  encodedData[20] & 0xF
```
Is this a db field? Perhaps it can be converted to the actual value, if we know how to do it; it'd be interesting to see how strongly the pod hears us.

Lytrix committed:
* Enum values in lowerCase
* DeliveryType as OptionSet
* Fix Typo original

Lytrix committed:
Added extendedBolus value to OptionSet DeliveryType,
to be the same as StatusError DeliveryType.

Lytrix committed:
Messageblock error: Function not closed with }

Lytrix committed:
Fixed typo original

Lytrix committed:
Added .none DeliveryType

Lytrix committed:
* Removed DeliveryInProgress error message handling
* Added none type
* Fixed typo original

Lytrix committed:
Added extra pod failure tests

Lytrix committed:
The statuslist of StatusError and StatusResponse is the same, so:
* Put all ReservoirStatus cases into StatusResponse
* Replaced ProgressType with ReservoirStatus in StatusError
